arctic/chunkstore/date_chunker.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd
+from pandas import Timestamp
+
+from ._chunker import Chunker
+from ..date import DateRange
+
+
+class DateChunker(Chunker):
+    def _get_date_chunk(self, date, chunk_size):
+        '''
+        format date appropriately for the chunk size
+
+        returns
+        -------
+        Formatted date string
+        '''
+        if chunk_size == 'Y':
+            return date.strftime('%Y')
+        elif chunk_size == 'M':
+            return date.strftime('%Y-%m')
+        elif chunk_size == 'D':
+            return date.strftime('%Y-%m-%d')
+
+    def _get_date_range(self, df):
+        """
+        get minx/max dates in the index of the dataframe
+
+        returns
+        -------
+        A tuple (start date, end date)
+        """
+        dates = df.index.get_level_values('date')
+        start = dates.min()
+        end = dates.max()
+        if isinstance(start, Timestamp):
+            start = start.to_pydatetime()
+        if isinstance(end, Timestamp):
+            end = end.to_pydatetime()
+        return start, end
+
+    def to_chunks(self, df, chunk_size):
+        """
+        chunks the dataframe/series by dates
+
+        returns
+        -------
+        generator that produces tuples: (start dt, end dt, dataframe/series)
+        """
+        if chunk_size not in ('D', 'M', 'Y'):
+            raise Exception("Chunk size must be one of D, M, Y")
```

Comment:
Year *nearly* fits 1min bars bars for most liquid markets.  I'd default to larger rather than smaller - so at least Month. For EOD data definitely year...